File: plugin-java/src/test/resources/com/sot/sonar/crypto/rules/samples/InsecureRandomSamples.java
package samples;

import java.security.SecureRandom;
import java.util.Random;

class InsecureRandomSamples {

    // --- Noncompliant: Random in security context (CRITICAL) ---

    String generateToken() {
        long token = new Random().nextLong(); // Noncompliant
        return Long.toHexString(token);
    }

    byte[] generateSalt() {
        byte[] salt = new byte[16];
        new Random().nextBytes(salt); // Noncompliant
        return salt;
    }

    String generateSessionId() {
        return String.valueOf(Math.random()); // Noncompliant
    }

    // --- Noncompliant: fixed-seed SecureRandom (CRITICAL, always) ---

    byte[] fixedSeedRandom() {
        SecureRandom sr = new SecureRandom(new byte[]{1, 2, 3}); // Noncompliant
        byte[] nonce = new byte[12];
        sr.nextBytes(nonce);
        return nonce;
    }

    // --- Compliant: no security context (skip) ---

    int rollDice() {
        return new Random().nextInt(6) + 1; // Compliant — game logic
    }

    double randomUiOffset() {
        return Math.random() * 100; // Compliant — UI, no security context
    }

    // --- Compliant: proper SecureRandom ---

    byte[] secureNonce() {
        SecureRandom sr = new SecureRandom();
        byte[] nonce = new byte[12];
        sr.nextBytes(nonce); // Compliant
        return nonce;
    }
}

File: plugin-java/src/test/resources/com/sot/sonar/crypto/rules/samples/InsecureRandomSamples.kt
package samples

import java.security.SecureRandom
import java.util.Random

// --- Noncompliant: Random in security context (CRITICAL) ---

fun generateToken(): String { // Noncompliant
    return java.lang.Long.toHexString(Random().nextLong())
}

fun generateSalt(): ByteArray { // Noncompliant
    val salt = ByteArray(16)
    Random().nextBytes(salt)
    return salt
}

fun generateSessionId(): String { // Noncompliant
    return Math.random().toString()
}

// --- Noncompliant: fixed-seed SecureRandom (CRITICAL, always) ---

fun fixedSeedRandom(): ByteArray { // Noncompliant
    val sr = SecureRandom(byteArrayOf(1, 2, 3))
    val nonce = ByteArray(12)
    sr.nextBytes(nonce)
    return nonce
}

// --- Compliant: no security context (skip) ---

fun rollDice(): Int { // Compliant — game logic
    return Random().nextInt(6) + 1
}

// --- Compliant: proper SecureRandom ---

fun secureNonce(): ByteArray { // Compliant
    val sr = SecureRandom()
    val nonce = ByteArray(12)
    sr.nextBytes(nonce)
    return nonce
}
